File: sdk/dart-holons/test/resolve_test.dart
import 'dart:io';

import 'package:holons/holons.dart';
import 'package:holons/src/discover.dart' as discover_impl;
import 'package:test/test.dart';

import 'test_support/discovery_fixture.dart';

void main() {
  final sdkRoot = Directory.current.path;

  tearDown(discover_impl.resetDiscoveryTestOverrides);

  group('resolve', () {
    test('known slug', () {
      final fixture = createRuntimeFixture(sdkRoot);
      addTearDown(() => deleteFixture(fixture));
      configureDiscoveryRuntime(fixture);

      writePackageHolon(
        Directory('${fixture.root.path}/alpha.holon'),
        slug: 'alpha',
        uuid: 'uuid-alpha',
      );

      final result =
          resolve(LOCAL, 'alpha', fixture.root.path, CWD, NO_TIMEOUT);

      expect(result.error, isNull);
      expect(result.ref?.info?.slug, equals('alpha'));
    });

    test('missing target', () {
      final fixture = createRuntimeFixture(sdkRoot);
      addTearDown(() => deleteFixture(fixture));
      configureDiscoveryRuntime(fixture);

      final result =
          resolve(LOCAL, 'missing', fixture.root.path, ALL, NO_TIMEOUT);

      expect(result.ref, isNull);
      expect(result.error, equals('holon "missing" not found'));
    });

    test('invalid specifiers', () {
      final fixture = createRuntimeFixture(sdkRoot);
      addTearDown(() => deleteFixture(fixture));
      configureDiscoveryRuntime(fixture);

      final result =
          resolve(LOCAL, 'alpha', fixture.root.path, 0xFF, NO_TIMEOUT);

      expect(result.ref, isNull);
      expect(result.error, contains('invalid specifiers'));
    });
  });
}

File: sdk/js-web-holons/test/resolve.test.mjs
import { describe, it } from "node:test";
import assert from "node:assert/strict";

import { ALL, LOCAL, NO_TIMEOUT, resolve } from "../src/index.mjs";

describe("resolve", () => {
    it("known slug returns not-found error", () => {
        const result = resolve(LOCAL, "gabriel-greeting-go", null, ALL, NO_TIMEOUT);
        assert.equal(result.ref, null);
        assert.equal(result.error, 'holon "gabriel-greeting-go" not found');
    });

    it("missing target returns not-found error", () => {
        const result = resolve(LOCAL, "", null, ALL, NO_TIMEOUT);
        assert.equal(result.ref, null);
        assert.equal(result.error, 'holon "" not found');
    });

    it("invalid specifiers returns error", () => {
        const result = resolve(LOCAL, "gabriel-greeting-go", null, 0xFF, NO_TIMEOUT);
        assert.equal(result.ref, null);
        assert.match(result.error || "", /invalid specifiers/i);
    });
});
